src/main/java/gov/nasa/jpl/mbee/mdk/tickets/BasicAuthAcquireTicketProcessor.java

```diff
@@ -0,0 +1,268 @@
+package gov.nasa.jpl.mbee.mdk.tickets;
+
+import java.awt.GridLayout;
+import java.awt.event.FocusAdapter;
+import java.awt.event.FocusEvent;
+import java.awt.event.HierarchyEvent;
+import java.awt.event.HierarchyListener;
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+
+import javax.swing.JButton;
+import javax.swing.JLabel;
+import javax.swing.JOptionPane;
+import javax.swing.JPanel;
+import javax.swing.JPasswordField;
+import javax.swing.JRootPane;
+import javax.swing.JTextField;
+import javax.swing.SwingUtilities;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.nomagic.magicdraw.core.Application;
+import com.nomagic.magicdraw.core.Project;
+import com.nomagic.task.ProgressStatus;
+import com.nomagic.ui.ProgressStatusRunner;
+
+import org.apache.http.client.methods.HttpRequestBase;
+
+import gov.nasa.jpl.mbee.mdk.http.ServerException;
+import gov.nasa.jpl.mbee.mdk.json.JacksonUtils;
+import gov.nasa.jpl.mbee.mdk.mms.MMSUtils;
+import gov.nasa.jpl.mbee.mdk.mms.actions.MMSLogoutAction;
+import gov.nasa.jpl.mbee.mdk.mms.endpoints.MMSEndpointBuilderConstants;
+import gov.nasa.jpl.mbee.mdk.mms.endpoints.MMSEndpointType;
+import gov.nasa.jpl.mbee.mdk.mms.endpoints.MMSLoginEndpoint;
+import gov.nasa.jpl.mbee.mdk.util.TicketUtils;
+import gov.nasa.jpl.mbee.mdk.util.Utils;
+
+public class BasicAuthAcquireTicketProcessor extends AbstractAcquireTicketProcessor {
+    public BasicAuthAcquireTicketProcessor(AbstractAcquireTicketProcessor processor) {
+        super(processor);
+    }
+
+    private static String username = "";
+    private static String password = "";
+
+    public static void clearPassword() {
+        password = "";
+    }
+
+    /**
+     * Method to set username and password for logging in to MMS. Can be called
+     * directly to pre-set the credential information for automation, but should be
+     * used with Utils.disablePopups(true) to prevent display of the standard log in
+     * window. Use without disabling popups will cause these values to be
```

Comment:
I don't believe this comment applies anymore.

Reply:
Comment should still apply as long as this class is included in the authentication chain